### DISCUSSION TITLE
Environment variables for Docker

### DISCUSSION THREAD

**Discussion by zmtk-josecardoso** [netbox-community/netbox#18251] [Help Wanted!]
Hello. I'm trying to install the docker image netboxcommunity/netbox:v4.1-3.0.2 and I can't get it working, I don't know why.
I'm trying to use some Enviroment Variables, to set an external PSQL database, also on docker. I've some some information outside the Docker Hub container image, but I don't know if the variables I'm using are correct.

![image](https://github.com/user-attachments/assets/61dfb5ba-67c0-4c49-83d3-296a172b4571)

Can someaone help please?

**Comment by candlerb** [comment]
"It doesn't work" is not a useful starting point. Does the container start? Does it terminate prematurely? What do the container logs show? What about logs on the postgres side?

You can see the environment variables used in [the source](https://github.com/netbox-community/netbox-docker/blob/3.0.2/configuration/configuration.py) and the [documentation](https://github.com/netbox-community/netbox-docker/wiki/Configuration).

You might also want to look at the [docker-compose.yml](https://github.com/netbox-community/netbox-docker/blob/3.0.2/docker-compose.yml) file and the [environment variables](https://github.com/netbox-community/netbox-docker/tree/3.0.2/env) it sets.

**Reply by zmtk-josecardoso** [reply]
@candlerb , thank's for the reply. You are absolutely right,
So, the container starts correctly and keeps running. My PG database is on another docker container. I've created the database, the user and granted all the necessary permissions to tne netbox db user. I've set both containers (netbox and postgres) on the same network and I can ping internally from both containers to each other. 
What "doesn't" work is that, when I connect to netbox web interface, nothing happens, just the "ERR_CONNECTION_REFUSED" error on the browser.

![image](https://github.com/user-attachments/assets/3c19e649-ba38-4195-81ec-a0a6477e186b)

![image](https://github.com/user-attachments/assets/1396487c-a759-4a7e-aeed-28786daada79)

That's why I suspect that there might be issues with my environment variables. Also, I'm probably doing something wrong, as I was expecting to have Redis installed, but there's no Redis container created.

So, that's what I'm passing through. I'll try to find some kind of step-by-step guide on the internet...

**Reply by candlerb** [reply]
You still haven't shown the container logs. Nor have you shown any of your Docker configuration. Nor even said what kind of Docker environment you have (there is a screenshot of some web UI?)

If nginx has started in the container, then I would guess you haven't forwarded the correct port to the container in your docker environment. nginx listens on port 8080 inside the container, but you can choose what port on the host to forward to the container.

https://github.com/netbox-community/netbox-docker/blob/3.0.2/docker/nginx-unit.json#L2-L8
https://github.com/netbox-community/netbox-docker/blob/3.0.2/docker-compose.override.yml.example#L3-L4

And indeed you will need to run redis. Like I say, you can follow the docker-compose configuration to find out all the pieces you need to run together. If you're going to assemble it yourself from parts, then you're on your own.

Good luck.
